flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/factories/TestValuesTableFactory.java

```diff
@@ -1883,6 +1892,26 @@ public LookupRuntimeProvider getLookupRuntimeProvider(LookupContext context) {
             }
         }
 
+        private List<Row> deduplicateDataByPk(List<Row> data) {
+            if (primaryKeyIndices.length == 0) {
+                return data;
+            }
+            // <pk, data>
+            LinkedHashMap<Row, Row> pkMap = new LinkedHashMap<>();
+            for (Row row : data) {
+                pkMap.put(extractPk(row), row);
```

Comment:
Do we need to check rowkind here? If the rowkind is DELETE, should we remove the row from the map?